appservice/src/createAppService/createWebApp.ts

```diff
@@ -30,6 +31,9 @@ export async function setWizardContextDefaults(wizardContext: IAppServiceWizardC
         } else if (await fse.pathExists(path.join(fsPath, 'requirements.txt'))) {
             // requirements.txt are used to pip install so a good way to determine it's a Python app
             wizardContext.recommendedSiteRuntime = LinuxRuntimes.python;
+        } else if (await javaUtils.isJavaProject(fsPath)) {
```

Comment:
You're setting things that result in fewer prompts, which should only be done `if (!advancedCreation)` (like below). `recommendedSiteRuntime` is unique because it moves runtimes to the top of the quick pick, but _still prompts the user_. Thus it can ignore `advancedCreation`.